src/ray/raylet/local_object_manager.cc

```diff
@@ -343,46 +343,23 @@ void LocalObjectManager::AddSpilledUrls(
   }
 }
 
+std::string LocalObjectManager::GetSpilledObjectURL(const ObjectID &object_id) {
+  auto entry = spilled_objects_url_.find(object_id);
+  if (entry != spilled_objects_url_.end()) {
+    return entry->second;
+  } else {
+    return "";
+  }
+}
+
 void LocalObjectManager::AsyncRestoreSpilledObject(
-    const ObjectID &object_id, const std::string &object_url, const NodeID &node_id,
+    const ObjectID &object_id, const std::string &object_url,
     std::function<void(const ray::Status &)> callback) {
   if (objects_pending_restore_.count(object_id) > 0) {
     // If the same object is restoring, we dedup here.
     return;
   }
 
-  if (is_external_storage_type_fs_ && node_id != self_node_id_) {
-    RAY_CHECK(!node_id.IsNil());
-    // If we know where this object was spilled, and the current node is not that one,
-    // send a RPC to a remote node that spilled the object to restore it.
-    RAY_LOG(DEBUG) << "Send an object restoration request of id: " << object_id
-                   << " to a remote node: " << node_id;
-    // TODO(sang): We need to deduplicate this remote RPC. Since restore request
-    // is retried every 10ms without exponential backoff, this can add huge overhead to
-    // a remote node that spilled the object.
-    restore_object_from_remote_node_(object_id, object_url, node_id);
-    if (callback) {
-      callback(Status::OK());
-    }
-    return;
-  }
-
-  // Restore the object.
-  RAY_LOG(DEBUG) << "Restoring spilled object " << object_id << " from URL "
-                 << object_url;
-  if (is_external_storage_type_fs_ && spilled_objects_url_.count(object_id) == 0) {
-    RAY_CHECK(!node_id.IsNil());
-    RAY_LOG(DEBUG)
-        << "Restoration request was ignored because the node didn't spill the object "
-        << object_id << " yet. It will be retried...";
-    // If the object wasn't spilled yet on this node, just return. The caller should retry
-    // in this case.
-    if (callback) {
-      callback(Status::OK());
-    }
-    return;
-  }
-
   RAY_CHECK(objects_pending_restore_.emplace(object_id).second)
```

Comment:
Local object manager still needs to be able to handle concurrent restoration requests from different nodes while guarding against duplicate restoration requests from the same node.

Reply:
Why do we need to handle concurrent restoration requests from different nodes differently?

Reply:
Ah never mind, we don't need to handle it differently since interested nodes join in waiting in the object manager, and the object added notification will trigger pushes for all interested nodes. Nice!